src/lib/zAuction.ts

```diff
@@ -87,8 +103,41 @@ export async function getBidsForNft(
 	return bids;
 }
 
+export async function getBidsListForNfts(
+	baseApiUri: string,
+	contract: string,
+	tokenId: string[],
+) {
+	const nftIdsArray: string[] = [];
+
+	for (let i = 0; i < tokenId.length; i++) {
+		// it builds the array of nftIds strings
+		nftIdsArray.push(getNftId(contract, tokenId[i]));
+	}
+
+	const nftIdsBody = { nftIds: nftIdsArray }; //builds the correct body to send
+
+	const endpoints = getApiEndpoints(baseApiUri);
+
+	const response = await fetch(endpoints.bidListEndpoint, {
+		method: 'POST',
+		headers: { 'Content-Type': 'application/json' },
+		body: JSON.stringify(nftIdsBody),
+	});
+
+	const jsonResponse = (await response.json()) as BidsListResponseDto[]; //awaits the json of the response
+
+	const bidsListMap = new Map <string, NftIdBidsOfListsDto[]>(); //mapping of tokenId as key and returned bids as value
+
+	for (let i = 0; i < tokenId.length; i++) { //populate the map with the data for every user
```

Comment:
You are iterating the `jsonResponse` array but using the `tokenId` array as the iterator. This is dangerous. There may be a bug in the zAuction API where `jsonResponse.length !== tokenId.length` and this code will break